inspirehep/modules/workflows/tasks/refextract.py

```diff
@@ -184,3 +189,46 @@ def extract_references_from_raw_ref(reference, custom_kbs_file=None):
     return extract_references_from_text(
         raw_ref['value'], source=raw_ref['source'], custom_kbs_file=custom_kbs_file
     )
+
+
+def match_references(references):
+    """Match references to record ids given a list of references.
+
+    Args:
+        references: Schema compliant list of references
+
+    Returns:
+        associated_references: The same list with the associated record ids (if any)
+    """
+
+    for i, reference in enumerate(references):
+        matched_recid = match_reference(reference)
+        if matched_recid:
+            references[i]['record'] = get_record_ref(matched_recid, 'literature')
+
+    return references
+
+
+def match_reference(reference):
+    """Match references given a reference metadata using InspireMatcher queires.
+
+    Args:
+        reference: The reference metadata
+        config: The configurtaion(s) for InspireMatcher queries
+
+    Returns:
+        The record ID of the matched reference
+    """
+
+    config_default = current_app.config['WORKFLOWS_REFERENCE_MATCHER_DEFAULT_CONFIG']
+    config_jcap_and_jhep = current_app.config['WORKFLOWS_REFERENCE_MATCHER_JHEP_AND_JCAP_CONFIG']
+
+    journal_title = get_value(reference, 'reference.publication_info.journal_title')
+    if journal_title in ['JCAP', 'JHEP']:
```

Comment:
it would be cleaner to have
```python
config = config_jcap_and_jhep if journal_title in ['JCAP', 'JHEP'] else config_default
result = next(match(reference, config_jcap_and_jhep), None)
if result:
    return result['_source']['control_number']
return None
```